src/qt/anybutton.cpp

```diff
@@ -24,18 +25,87 @@ class wxQtPushButton : public wxQtEventSignalHandler< QPushButton, wxAnyButton >
 {
 
 public:
+    virtual bool eventFilter(QObject *watched, QEvent *event);
     wxQtPushButton( wxWindow *parent, wxAnyButton *handler);
 
 private:
     void clicked(bool);
+    wxAnyButton *m_button;
 };
 
 wxQtPushButton::wxQtPushButton(wxWindow *parent, wxAnyButton *handler)
     : wxQtEventSignalHandler< QPushButton, wxAnyButton >( parent, handler )
 {
+    m_button = handler;
+    installEventFilter( this );
     connect(this, &QPushButton::clicked, this, &wxQtPushButton::clicked);
 }
 
+bool wxQtPushButton::eventFilter(QObject *watched, QEvent *event)
+{
+// TODO: Disabled control shouldn't get any events
+    QPushButton *button = qobject_cast<QPushButton *>( watched );
+    if( !button )
```

Comment:
Isn't `button` the same as `this`? In which case testing for `watched != this` should be enough.

Reply:
I'l check...